From 56abdf56024a46a2ea09a72a11a85f3bcdc61fd5 Mon Sep 17 00:00:00 2001
From: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
Date: Thu, 27 Feb 2025 20:16:53 -0300
Subject: [PATCH] ceph-csi-cephfs: add default fields to values.yaml

This patch adds fields and values necessary for the proper use
of the cephfs chart in the system.

Signed-off-by: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
---
 charts/ceph-csi-cephfs/values.yaml | 41 ++++++++++++++++++++++++++++--
 1 file changed, 39 insertions(+), 2 deletions(-)

diff --git a/charts/ceph-csi-cephfs/values.yaml b/charts/ceph-csi-cephfs/values.yaml
index f30e2b7..160b7da 100644
--- a/charts/ceph-csi-cephfs/values.yaml
+++ b/charts/ceph-csi-cephfs/values.yaml
@@ -290,10 +290,10 @@ provisioner:

   podAnnotations: {}

-# readAffinity:
+readAffinity:
 # Enable read affinity for CephFS subvolumes. Recommended to
 # set to true if running kernel 5.8 or newer.
-# enabled: false
+  enabled: false
 # Define which node labels to use as CRUSH location.
 # This should correspond to the values set in the CRUSH map.
 # NOTE: the value here serves as an example
@@ -375,6 +375,26 @@ storageClass:
   # mountOptions:
   #   - discard

+  # Ceph user name to access this pool
+  userId: kube
+  # K8 secret name with key for accessing the Ceph pool
+  userSecretName: ceph-secret-kube
+  # Pool replication
+  replication: 1
+  # Pool min replication
+  min_replication: 1
+  # Pool crush rule name
+  crush_rule_name: storage_tier_ruleset
+  # Pool chunk size / PG_NUM
+  chunk_size: 8
+  # Additional namespace to allow storage class access (other than where
+  # installed)
+  additionalNamespaces:
+  - default
+  - kube-public
+  # Ceph pools name
+  metadata_pool: kube-cephfs-metadata
+
 secret:
   # Specifies whether the secret should be created
   create: false
@@ -431,6 +451,23 @@ externallyManagedConfigmap: false
 cephConfConfigMapName: ceph-config
 # Name of the configmap used for encryption kms configuration
 kmsConfigMapName: ceph-csi-encryption-kms-config
+
+#
+# Defaults for storage classes.
+#
+classdefaults:
+  # Define ip addresses of Ceph Monitors
+  monitors:
+  - 192.168.204.2:6789
+  # K8 secret name for the admin context
+  adminId: admin
+  adminSecretName: ceph-secret-admin
+  cephFSNamespace: kube-system
+
+storage_init:
+  image:
+    repository: docker.io/openstackhelm/ceph-config-helper
+    tag: ubuntu_jammy_19.2.1-1-20250207
 # CephFS RadosNamespace used to store CSI specific objects and keys.
 # radosNamespaceCephFS: csi
 # Unique ID distinguishing this instance of Ceph CSI among other instances,
--
2.34.1
